File: src/main/scala/com/github/tetrisanalyzer/piece/PieceAny.scala
package com.github.tetrisanalyzer.piece

object PieceAny {
  val Number = 8.toByte
}

class PieceAny extends Piece {
  val number = PieceAny.Number
  val character = 'x'
  protected val widths = Array.empty[Int]
  protected val heights = Array.empty[Int]
  protected val shapes = Array.empty[PieceShape]
}

File: langs/java/src/main/java/com/github/tetrisanalyzer/piece/PieceAny.java
package com.github.tetrisanalyzer.piece;

import com.github.tetrisanalyzer.settings.PieceSettings;

public class PieceAny extends Piece {

    public PieceAny(PieceSettings settings) {
        super(settings);
    }

    @Override public byte number() { return 8; }
    @Override public char character() { return 'x'; }
    @Override protected int[] widths() { return new int[0]; }
    @Override protected int[] heights() { return new int[0]; }
    @Override protected PieceShape[] shapes() { return new PieceShape[0]; }
}
